examples/tv-casting-app/android/App/app/src/main/jni/com/matter/casting/core/CastingApp.java

```diff
@@ -0,0 +1,163 @@
+      return MatterError.CHIP_ERROR_INCORRECT_STATE;
+    }
+
+    this.appParameters = appParameters;
+    this.nsdManagerResolverAvailState =
+        new NsdManagerServiceResolver.NsdManagerResolverAvailState();
+
+    Context applicationContext = appParameters.getApplicationContext();
+    AndroidChipPlatform chipPlatform =
+        new AndroidChipPlatform(
+            new AndroidBleManager(),
+            new PreferencesKeyValueStoreManager(appParameters.getApplicationContext()),
+            appParameters.getConfigurationManagerProvider().get(),
+            new NsdManagerServiceResolver(applicationContext, nsdManagerResolverAvailState),
+            new NsdManagerServiceBrowser(applicationContext),
+            new ChipMdnsCallbackImpl(),
+            new DiagnosticDataProviderImpl(applicationContext));
+
+    CommissionableData commissionableData = appParameters.getCommissionableDataProvider().get();
+    boolean updated =
+        chipPlatform.updateCommissionableDataProviderData(
+            commissionableData.getSpake2pVerifierBase64(),
+            commissionableData.getSpake2pSaltBase64(),
+            commissionableData.getSpake2pIterationCount(),
+            commissionableData.getSetupPasscode(),
+            commissionableData.getDiscriminator());
+    if (!updated) {
+      Log.e(
+          TAG, "CastingApp.initApp failed to updateCommissionableDataProviderData on chipPlatform");
+      return MatterError.CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    MatterError err = finishInitialization(appParameters);
+
+    if (err.hasNoError()) {
+      chipAppServer = new ChipAppServer(); // get a reference to the Matter server now
+      mState = CastingAppState.NOT_RUNNING; // initialization done, set state to NOT_RUNNING
+    }
+    return err;
+  }
+
+  /**
+   * Starts the Matter server that the CastingApp runs on and registers all the necessary delegates
+   */
+  public MatterError start() {
+    Log.i(TAG, "CastingApp.start called");
+    if (mState != CastingAppState.NOT_RUNNING) {
+      return MatterError.CHIP_ERROR_INCORRECT_STATE;
+    }
+
+    boolean serverStarted = chipAppServer.startApp();
+    if (!serverStarted) {
+      Log.e(TAG, "CastingApp.start failed to start Matter server");
+      return MatterError.CHIP_ERROR_INCORRECT_STATE;
+    }
+
+    MatterError err = finishStartup();
+    if (err.hasNoError()) {
+      mState = CastingAppState.RUNNING; // CastingApp started successfully, set state to RUNNING
+    }
+    return err;
+  }
+
+  /**
+   * Stops the Matter server that the CastingApp runs on
+   *
+   * @return
+   */
+  public MatterError stop() {
+    Log.i(TAG, "CastingApp.stop called");
+    if (mState != CastingAppState.RUNNING) {
+      return MatterError.CHIP_ERROR_INCORRECT_STATE;
+    }
+
+    boolean serverStopped = chipAppServer.stopApp();
+    if (!serverStopped) {
+      Log.e(TAG, "CastingApp.stop failed to stop Matter server");
+      return MatterError.CHIP_ERROR_INCORRECT_STATE;
```

Comment:
is this the most accurate error for this scenario?

Reply:
Given that the underlying ChipAppServer in the core SDK that we are using here, is returning a hardcoded boolean (JNI\_TRUE), CHIP\_ERROR\_INCORRECT\_STATE seemed appropriate to return.